### PR TITLE
[NCL-7944] Add ability to skip callback

### PR DESCRIPTION
**Pull request by thescouser89**
We can just use the data provided in the initial request. This is done by specifying a custom configuration where skipCallback is set to true.

### CODE REVIEW

File: dto/src/main/java/org/jboss/pnc/rex/dto/ConfigurationDTO.java [hunk 1]
```diff
@@ -33,5 +33,6 @@
 @ToString
 public class ConfigurationDTO {
 
-    public boolean passResultsOfDependencies = false;
+    private boolean passResultsOfDependencies = false;
+    private boolean skipCallback = false;
```
Comment (michalovjan) [inline dto/src/main/java/org/jboss/pnc/rex/dto/ConfigurationDTO.java:37]: could you rename this to fastStart or skipStartCallback?

**Review by michalovjan — CHANGES_REQUESTED** [review]
1. Could you implement the same functionality for Cancellation?
2. Can you omit the callback(make it null/not put to JSON at all) in StartRequest/StopRequest if the callback should be skipped?

File: dto/src/main/java/org/jboss/pnc/rex/dto/ConfigurationDTO.java (hunk 1 above)
Comment (thescouser89) [reply dto/src/main/java/org/jboss/pnc/rex/dto/ConfigurationDTO.java:37]: yup sure!

**Comment by thescouser89** [comment]
> 1.Could you implement the same functionality for Cancellation?
Yup I'll look into it

> 2. Can you omit the callback(make it null/not put to JSON at all) in StartRequest/StopRequest if the callback should be skipped?

Will look into it. Thanks for the suggestions!

File: dto/src/main/java/org/jboss/pnc/rex/dto/ConfigurationDTO.java (hunk 1 above)
Comment (thescouser89) [reply dto/src/main/java/org/jboss/pnc/rex/dto/ConfigurationDTO.java:37]: The `skipStartCallback` naming won't work if I also include this for cancellation.

**Comment by thescouser89** [comment]
@michalovjan  I added `StartRequestWithCallback` and `StopRequestWithCallback`. I was debating whether to have `StartRequestWithoutCallback` instead but I think having a negation would be confusing. Let me know what you think! :)

**Comment by michalovjan** [comment]
> :(

sorryy, thanks for reminding my stupid forgetful head

File: dto/src/main/java/org/jboss/pnc/rex/dto/ConfigurationDTO.java (hunk 1 above)
Comment (michalovjan) [inline dto/src/main/java/org/jboss/pnc/rex/dto/ConfigurationDTO.java:37]: can you split the configuration for start and cancel? I can see someone would want fast cancel but regular start with callback

File: core/src/main/java/org/jboss/pnc/rex/core/TaskControllerImpl.java [hunk 2]
```diff
@@ -207,8 +207,13 @@ private Transition getTransition(Task task) {
                 if (task.getStopFlag() == StopFlag.CANCELLED)
                     yield Transition.STARTING_to_STOP_REQUESTED;
                 List<ServerResponse> responses = task.getServerResponses().stream().filter(sr -> sr.getState() == State.STARTING).toList();
-                if (responses.stream().anyMatch(ServerResponse::isPositive))
-                    yield Transition.STARTING_to_UP;
+                if (responses.stream().anyMatch(ServerResponse::isPositive)) {
```
Comment (michalovjan) [inline core/src/main/java/org/jboss/pnc/rex/core/TaskControllerImpl.java:210]: for cancellation equivalent? I guess also new Transition STOP-REQUESTED -> STOPPED

**Review by michalovjan — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/jboss/pnc/rex/core/TaskControllerImpl.java (hunk 2 above)
Comment (thescouser89) [reply core/src/main/java/org/jboss/pnc/rex/core/TaskControllerImpl.java:218]: Thanks for the great suggestion!! I completely omitted the stopCallback option. Added in the last commit :)

**Review by michalovjan — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/jboss/pnc/rex/core/TaskControllerImpl.java [hunk 3]
```diff
@@ -215,8 +215,13 @@ private Transition getTransition(Task task) {
...
+                    if (task.getConfiguration() != null && task.getConfiguration().isSkipStartRequestCallback()) {
+                        yield Transition.STARTING_to_SUCCESSFUL;
+                    } else {
+                        yield Transition.STARTING_to_UP;
+                    }
+                }
                 if (responses.stream().anyMatch(ServerResponse::isNegative))
```
[the remark this answers is not in the document]
Comment (thescouser89) [reply core/src/main/java/org/jboss/pnc/rex/core/TaskControllerImpl.java:225]: Yeah I think changing the ServerResponse.positive to an enum sounds good. I'll get to work and we'll get that PR merged! :muscle: 

I'll look more into how Rex generates its ServerResponse before I work more on this.

I am fine with using the status codes being considered START_FAILED or regular FAILED.
 
Noted also that this is also present for STOPs.

**Pull request closed without merging**